File: hydrograph.engine/hydrograph.engine.cascading/src/main/java/hydrograph/engine/adapters/RunProgramAdapter.java
package hydrograph.engine.adapters;

import hydrograph.engine.adapters.base.CommandAdapterBase;
import hydrograph.engine.commandtype.component.BaseCommandComponent;
import hydrograph.engine.commandtype.component.RunProgram;
import hydrograph.engine.core.component.generator.RunProgramEntityGenerator;
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent;

public class RunProgramAdapter extends CommandAdapterBase {

	
	private static final long serialVersionUID = 4972095665271276000L;
	BaseCommandComponent runProgram;
	RunProgramEntityGenerator entityGenerator;

	public RunProgramAdapter(TypeBaseComponent typeCommandComponent) {

		entityGenerator = new RunProgramEntityGenerator(typeCommandComponent);

	}

	@Override
	public BaseCommandComponent getComponent() {

		try {
			runProgram = new RunProgram(entityGenerator.getEntity());
		} catch (Throwable e) {
			throw new RuntimeException("Assembly Creation Error for RunProgram Component : " + e.getMessage());

		}
		return runProgram;
	}

}

File: hydrograph.engine/hydrograph.engine.spark/src/main/scala/hydrograph/engine/spark/components/adapter/RunProgramAdapter.scala
package hydrograph.engine.spark.components.adapter

import hydrograph.engine.core.component.generator.RunProgramEntityGenerator
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent
import hydrograph.engine.spark.components.RunProgramComponent
import hydrograph.engine.spark.components.adapter.base.RunProgramAdapterBase
import hydrograph.engine.spark.components.base.{CommandComponentSparkFlow}
import hydrograph.engine.spark.components.platform.BaseComponentParams

class RunProgramAdapter(typeBaseComponent: TypeBaseComponent) extends RunProgramAdapterBase{

  private  var runProgramEntityGenerator:RunProgramEntityGenerator=null
  private var runProgramComponent:RunProgramComponent=null

  override def createGenerator(): Unit = {
    runProgramEntityGenerator=new RunProgramEntityGenerator(typeBaseComponent)
  }

  override def createComponent(baseComponentParams: BaseComponentParams): Unit = {
    runProgramComponent = new RunProgramComponent(runProgramEntityGenerator.getEntity)
  }

  override def getComponent(): CommandComponentSparkFlow = runProgramComponent
}
